environs/manual/sshprovisioner/provisioner.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2016 Canonical Ltd.
+// Copyright 2016 Cloudbase Solutions SRL
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshprovisioner
+
+import (
+	"github.com/juju/loggo"
+
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/environs/manual"
+)
+
+var (
+	logger = loggo.GetLogger("juju.environs.manual.sshprovisioner")
+)
+
+const Scope = "ssh"
+
+type provisioner struct {
+}
+
+// New returns a new linux provision
+func New() *provisioner {
+	return &provisioner{}
+}
+
+// Provision returns a new machineId and nil if the provision process is done successfully
+// The func will manual provision a linux machine using as it's default protocol SSH
+func (provisioner) ProvisionMachine(args manual.ProvisionMachineArgs) (machineId string, err error) {
```

Comment:
As mentioned elsewhere, please make this a top-level function `func ProvisionMachine`.